pkg/controller/projectreference/projectreference_adapter.go

```diff
@@ -469,6 +480,26 @@ func (r *ReferenceAdapter) deleteCredentials() error {
 	return nil
 }
 
+func (r *ReferenceAdapter) checkAvailibilityZones() error {
```

Comment:
I think this function should be called ensureAvailabilityZonesSet() or sometthing similar. Also I think it should return if it modified the object or not. Read comment above for why.